dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -0,0 +1,358 @@
+﻿#include "pch.h"
+#include "AppNotificationManager.h"
+#include "Microsoft.Windows.AppNotifications.AppNotificationManager.g.cpp"
+#include <winrt/Windows.ApplicationModel.Core.h>
+#include <winrt/Windows.Storage.h>
+#include <winrt/base.h>
+#include "externs.h"
+#include "PushNotificationUtility.h"
+#include "AppNotificationUtility.h"
+#include <frameworkudk/pushnotifications.h>
+#include <frameworkudk/toastnotifications.h>
+#include <FrameworkUdk/toastnotificationsrt.h>
+#include "NotificationProperties.h"
+#include "NotificationTransientProperties.h"
+#include "AppNotification.h"
+#include <NotificationProgressData.h>
+#include <algorithm>
+#include <winerror.h>
+#include <string_view>
+#include <winrt/Windows.Foundation.Collections.h>
+
+using namespace std::literals;
+
+constexpr std::wstring_view expectedAppServerArgs = L"----AppNotificationActivated:"sv;
+
+namespace winrt
+{
+    using namespace winrt::Windows::UI;
+    using namespace winrt::Windows::Foundation;
+    using namespace winrt::Windows::Foundation::Collections;
+    using namespace winrt::Microsoft::Windows::AppNotifications;
+	using namespace Windows::ApplicationModel::Core;
+}
+
+namespace ToastABI
+{
+    using namespace ::ABI::Microsoft::Internal::ToastNotifications;
+    using namespace ::ABI::Windows::Foundation::Collections;
+}
+
+namespace PushNotificationHelpers
+{
+    using namespace winrt::Microsoft::Windows::PushNotifications::Helpers;
+}
+
+using namespace Microsoft::Windows::AppNotifications::Helpers;
+
+namespace winrt::Microsoft::Windows::AppNotifications::implementation
+{
+    winrt::Microsoft::Windows::AppNotifications::AppNotificationManager AppNotificationManager::Default()
+    {
+        static auto appNotificationManager{winrt::make<AppNotificationManager>()};
+        return appNotificationManager;
```

Comment:
This PR addresses this issue:  https://github.com/microsoft/WindowsAppSDK/pull/2187

Reply:
We need to use the COM static store and compare the singleton against that. Otherwise I believe we run into cleanup problems in the termination stage. Pauls change addresses that.